### ISSUE TITLE
pi0-fintune-performance

### ISSUE THREAD

**Issue by yanghb1** [Physical-Intelligence/openpi#427]
I have been fine-tuning the provided pi0-base model on my dataset using LeRobot. After training for 100,000 steps, I found that the model performs well on tasks that appeared in my dataset, but its performance on unseen tasks is very poor. It seems to lack the generalization ability of a VLA model. Is this phenomenon normal? Are there any strategies to improve this situation?

**Comment by uzhilinsky** [comment]
We've certainly been able to successfully fine-tune pi0-base on novel tasks. Hard to say what's going on in your case without having more context.

Just in case, are you using the right [norm stats](https://github.com/Physical-Intelligence/openpi/blob/36dc3c037eb8a3921be9ecb94369d60cbf56f58f/docs/norm_stats.md#normalization-statistics)?

**Comment by yanghb1** [comment]
Thank you for your patient response. May the sunshine bathe you. I am using pi0-base for training and fine-tuning on the lerobot project. The base model is from https://huggingface.co/lerobot/pi0. My training dataset contains 700 collected trajectories with only one task: opening a cabinet door. After training for 100,000 steps, the loss has converged to 0.03. During inference, I found that the model can perform the task of opening the cabinet door as in the dataset, but it performs poorly on other instructions, such as grabbing a ballpoint pen or moving the yellow tape to the left of the blue tape, and still tends to open the cabinet door. Is this phenomenon normal? Are there any other ways to improve this? I read in the paper that I thought fine-tuning the expert model and state projector with local data was sufficient for executing various instructions on a local robotic arm.

**Comment by uzhilinsky** [comment]
100,000 steps is a lot for such a small dataset. It's very likely that the model has overfit to your data.

Note that we are using 20K or 30K in our fine-tuning examples with much larger datasets. Have you tried training with fewer steps?

**Comment by oxFFFF-Q** [comment]
Same issue here. Did you solve it?

**Comment by YanJiaHuan** [comment]
anyone tried they said? 
With larger dataset and smaller training steps.
I tried 200 demos (10-15s each,10hz) with over 300k steps of training, the model performs terrible, simple pick&place task.

**Comment by FanZhang91** [comment]
> anyone tried they said? With larger dataset and smaller training steps. I tried 200 demos (10-15s each,10hz) with over 300k steps of training, the model performs terrible, simple pick&place task.

did you solve this question?

**Comment by YanJiaHuan** [comment]
now can work when >400 demos(simple pick&place task, which can also work when using Diffusion Policy), we simply optimized our inference codes.

**Comment by FishofJLlake** [comment]
Can I perform further fine-tuning on the π0-DROID model using my own dataset? Compared to fine-tuning based on the pi0-base model, can I expect additional benefits? Or, going even further, is it possible to perform joint training with the DROID model and my own dataset?

**Comment by AricLau07** [comment]
> now can work when >400 demos(simple pick&place task, which can also work when using Diffusion Policy), we simply optimized our inference codes.

Hi! Great to hear it works with 400+ demos. Just curious : how many training steps did you use for fine-tuning?

**Comment by xiaoen0** [comment]
> now can work when >400 demos(simple pick&place task, which can also work when using Diffusion Policy), we simply optimized our inference codes.现在可以在 >400 个演示时工作（简单的拾取和放置任务，在使用扩散策略时也可以工作），我们只是优化了我们的推理代码。

Could you please share what optimizations were made?

**Comment by wjydym** [comment]
> now can work when >400 demos(simple pick&place task, which can also work when using Diffusion Policy), we simply optimized our inference codes.

Could you please share what optimizations were made?
